src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs

```diff
@@ -0,0 +1,137 @@
+﻿using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.CodeCoverage
+{
+    public static class CodeCoverageUtilities
+    {
+        public const string RawFilesDirectory = "Code Coverage Files";
+        public const string ReportDirectory = "Code Coverage Report";
+        public const string SummaryFileDirectory = "summary";
+        public const string DefaultIndexFile = "index.html";
+
+        public static void CopyFilesFromFileListWithDirStructure(List<string> files, ref string destinatonFilePath)
+        {
+            string commonPath = null;
+            if (files != null)
+            {
+                files.RemoveAll(q => q == null);
+
+                if (files.Count > 1)
+                {
+                    files.Sort();
+                    commonPath = SharedSubstring(files[0], files[files.Count - 1]);
+                }
+
+                foreach (var file in files)
+                {
+                    string newFile = null;
+
+                    if (!string.IsNullOrEmpty(commonPath))
+                    {
+                        newFile = file.Replace(commonPath, "");
+                    }
+                    else
+                    {
+                        newFile = Path.GetFileName(file);
+                    }
+
+                    newFile = Path.Combine(destinatonFilePath, newFile);
+                    Directory.CreateDirectory(Path.GetDirectoryName(newFile));
+                    File.Copy(file, newFile, true);
+                }
+            }
+        }
+
+        public static XmlDocument ReadSummaryFile(IExecutionContext context, string summaryXmlLocation)
+        {
+            string xmlContents = "";
+
+            //read xml contents
+            if (!File.Exists(summaryXmlLocation))
+            {
+                throw new ArgumentException(StringUtil.Loc("FileDoesNotExist", summaryXmlLocation));
+            }
+
+            xmlContents = File.ReadAllText(summaryXmlLocation);
```

Comment:
We have file exists check just above. If there is any exception in reading file it will be thrown